src/pkg/cluster/injector.go

```diff
@@ -432,66 +436,61 @@ func (c *Cluster) buildInjectionPod(node, image string, payloadConfigmaps []stri
 	return pod, nil
 }
 
-// GetImagesFromAvailableNodes checks for images on schedulable nodes within a cluster and returns
-func (c *Cluster) getImagesAndNodesForInjection(timeoutDuration time.Duration) (imageNodeMap, error) {
-	timeout := time.After(timeoutDuration)
+// getImagesAndNodesForInjection checks for images on schedulable nodes within a cluster.
+func (c *Cluster) getImagesAndNodesForInjection(ctx context.Context) (imageNodeMap, error) {
 	result := make(imageNodeMap)
 
+	timer := time.NewTimer(0)
```

Comment:
This is an interesting thing todo. Is the purpose of this and the reset further down to make sure that the timer is immediate?

Reply:
Yes, exactly. Using `time.NewTimer(0)` lets the timer start right away. Originally, we used `time.Sleep()`, but it's a blocking operation, which isn't ideal. I then switched to `time.After()` for better responsiveness since it doesn't block other operations. @AustinAbro321 suggested switching to `time.NewTimer` and it seems to work well for this use case

Reply:
TODO is supposed to be a placeholder, in tests we can make a context and pass it through the functions to ensure we are tests the functions as close as they'll be used normally. In tests I have seen it more common to make a context, or pass a background.